internal/controller/pkg/revision/reconciler.go

```diff
@@ -54,6 +54,8 @@ import (
 
 const (
 	reconcileTimeout = 3 * time.Minute
+	// the max size of a package parsed by the parser
+	maxPackageSize = 100 << 20 // 100 MB
```

Comment:
did you already take a look at the size of the biggest packages already in https://marketplace.upbound.io/? curious as to how close we already are to this limit.  with a full controller environment for providers, i can imagine we're not that far away from already bumping into this 🤔

Reply:
what do you mean by "full controller environment for providers"

Reply:
BTW, not sure it's clear, but it's a 100MB of YAMLs, not the size of the whole image

Reply:
wrote a small test checking the size for a few images: 

- xpkg.upbound.io/upbound/platform-ref-aws:v0.6.0: 0.04052734375MB
- xpkg.upbound.io/upbound/platform-ref-aws:latest: 0.03369140625MB
- xpkg.upbound.io/upbound/platform-ref-azure:v0.7.0: 0.03173828125MB
- xpkg.upbound.io/upbound/platform-ref-azure:latest: 0.02197265625MB
- xpkg.upbound.io/upbound/platform-ref-gcp:v0.4.0: 0.02734375MB
- xpkg.upbound.io/upbound/platform-ref-gcp:latest: 0.0224609375MB

As you can see all of them are actually 30KB, so well below the limit of 100MB

Reply:
Monolithic providers are indeed quite big, still way below 100MB: 
- xpkg.upbound.io/upbound/provider-aws:v0.37.0: 25.2275390625MB, annotation: true
- xpkg.upbound.io/upbound/provider-gcp:v0.34.0: 11.8583984375MB, annotation: true
- xpkg.upbound.io/upbound/provider-azure:v0.34.0: 23.216796875MB, annotation: true

we could maybe make this configurable via a flag, WDYT @jbw976 @turkenh?

Reply:
Is that the size of just the yaml layer @phisco or is that the entire OCI image?

I'd prefer to keep the complexity and moving parts here very slim and not add a configurable flag/machinery 🤓 

to give us a bit more room though, maybe we can bump the max value to something a bit higher, e.g. 256MB? what is the value at which we'd actually be concerned about a denial of service attack happening?

You can make the final call here @phisco, and @turkenh has already approved ✅

Reply:
YAML layer only, I'll increase this to 200MB just to play safe and merge it 👍